src/java/main/org/apache/zookeeper/server/quorum/Learner.java

```diff
@@ -364,6 +364,7 @@ protected void syncWithLeader(long newLeaderZxid) throws Exception{
         readPacket(qp);
         LinkedList<Long> packetsCommitted = new LinkedList<Long>();
         LinkedList<PacketInFlight> packetsNotCommitted = new LinkedList<PacketInFlight>();
+        boolean syncSnapshot = false;
```

Comment:
Another possibility as I just commented is to get rid of this variable and always Fsync snapshot serialization.